server/args/Cargo.toml

```diff
@@ -0,0 +1,10 @@
+[package]
+name = "umi-server-args"
+description = "Umi execution HTTP server settings"
+version = "0.1.0"
+edition = "2021"
+
+[dependencies]
+toml.workspace = true
+optional_struct.workspace = true
```

Comment:
I don't really like adding a crate dependency that is preventing only a few lines of boilerplate. Supply chain attacks can happen, especially with crates that are not widely used.

I would rather we explicitly have one type for the config taken from environment (whether that be disk, command line or env vars) where all the fields are optional, and a separate type for the "logical" config where none of the fields are optional. This crate provides this functionality, but ties the two types together via a proc macro whereas I think it might be useful to allow the two types to evolve independently since it may not always be true that the logical config fields correspond 1:1 with the environmental config fields.